src/metadata_validator.py

```diff
@@ -151,23 +152,35 @@ def validate(self, submission_id, scope):
     def validate_nodes(self, data_records):
         #2. loop through all records and call validateNode
         updated_records = []
+        qc_results = []
         validated_count = 0
         try:
             for record in data_records:
+                qc_result = get_qc_result(record, self.submission, VALIDATION_TYPE_METADATA, self.mongo_dao)
                 status, errors, warnings = self.validate_node(record)
-                
                 if errors and len(errors) > 0:
                     self.isError = True
-                    record[ERRORS] = errors
+                    # record[ERRORS] = errors
+                    qc_result[ERRORS] = errors
+                    qc_result["severity"] = "Error"
                 else:
-                    record[ERRORS] = []
+                    # record[ERRORS] = []
+                    qc_result[ERRORS] = []
                 if warnings and len(warnings)> 0: 
                     self.isWarning = True
-                    record[WARNINGS] = warnings
+                    # record[WARNINGS] = warnings
+                    qc_result[WARNINGS] = warnings
+                    qc_result["severity"] = "Warning"
```

Comment:
Please use appropriate constant for "Warning"

Reply:
Updated.  Please continue.